hugo/content/docs/langium-overview.md

```diff
@@ -0,0 +1,103 @@
+
+```ts
+Person:
+    'person' // keyword 
+    name=ID // semantic assignment
+;
+```
+
+Parses a simple object `{ name: 'John' }`. To interact with this object in a type safe manner, the `langium-cli` tool generates TypeScript type definitions from its parser rule. The `Person` parser rule generates the following interface:
+
+```ts
+interface Person {
+    name: String
+}
+```
+
+These interfaces allow you to safely traverse your abstract syntax tree. In case your grammar changes, they will also notify you of any breaking changes which might require you to change your domain logic.
+
+## Cross References and Linking
+
+To express any kind of relationship between elements in your language, you will need to **reference** them.
+The process of resolving these references, i.e. identifying what element of your language hides behind a certain name, is called _linking_.
+Performing the linking process in a deterministic manner with a lot of objects in your project requires sound linking design.
+
+Langium accomplishes this feat by using the concept of 'scoping'. You likely know scopes from programming, where some variables are only available from certain scopes:
+
+```ts
+let x = 42;
+x = 3; // References the `x` defined in the previous line
+```
+
+```ts
+if (something) {
+    let y = 42;
+}
+y = 3; // Cannot link, `y` isn't in any of the available scopes
+```
+
+The same occurs in Langium. To enable more complex scoping behavior, you can create a custom domain scoping. For example, common object-oriented languages need a more involved scoping mechanism to resolve references to fields and methods of a class:
+
+```ts
+class X {
+    y(): void { ... }
+}
+
+const instance = new X(); // Symbol `X` is in the local scope
+instance.y(); // Symbol `y` exists in the scope of the `X` class
+```
+
+Once your domain specific scoping rules have been defined, Langium will take care of linking by itself, reporting any errors appearing 
+
+## Workspace Management
+
+Like with common modularized programming languages, domain logic written in your DSL will usually be split across multiple files to facilitate ease of use and maintenance efforts. This is also possible using Langium, which automatically tries to pick up any files belonging to your current project.
+
+When running a Langium based language in a [language server](https://microsoft.github.io/language-server-protocol/), all files in your workspace, i.e. the folder containing your current project, matching your DSL Id will automatically be picked up and processed. In addition, any changes in your workspace will be handled as well. Dealing with added, changed or deleted files in a workspace with multiple hundreds of files can become complicated and decrease performance drastically if not done correctly. Langium employs heuristics to only invalidate and recompute what is actually necessary.
```

Comment:
suggestion to use parentheses:

> ... ,i.e. the folder containing your current project, ...

... **(**the folder containing your current project**)** ...

Reply:
Would be helpful to be clearer about what 'DSL Id' is referring to. Is it the associated file extension(s)?

Reply:
suggestion, assuming this is based on what files have been changed?

> ... what is actually necessary...

what **files have changed**.

Reply:
>assuming this is based on what files have been changed?

Not exactly, most of the times we actually have to invalidate more parts of the workspace (which is where the smart heuristics come into play), since we need to invalidate all files which have references to the changed files, as these references may no longer be valid.